ipmininet/install/build_openr-rc-20190419-11514.sh

```diff
@@ -0,0 +1,301 @@
+#!/bin/bash
+
+set -exo pipefail
+### Install packages for Debian-based OS ###
+
+apt-get update && apt-get install -yq autoconf-archive bison build-essential cmake curl flex git gperf joe libboost-all-dev libcap-dev libdouble-conversion-dev libevent-dev libgflags-dev libgoogle-glog-dev libkrb5-dev libpcre3-dev libpthread-stubs0-dev libnuma-dev libsasl2-dev libsnappy-dev libsqlite3-dev libssl-dev libtool netcat-openbsd pkg-config sudo unzip wget python3-venv python-setuptools python3-setuptools python-pip python3-pip ccache
+apt-get install -yq gcc-'5' g++-'5'
+update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-'5' 40 --slave /usr/bin/g++ g++ /usr/bin/g++-'5'
+update-alternatives --config gcc
+
+PY_MAJOR=${PY_MAJOR:-2}
+
+if [ "$PY_MAJOR" -ne 2 ] && [ "$PY_MAJOR" -ne 3 ]; then
+    >&2 echo "PY_MAJOR must be set to either 2 or 3."
+    exit 1
+fi
+
+PY_BIN="python$PY_MAJOR"
+PIP_BIN="pip$PY_MAJOR"
+
+PY_VERSION="$($PY_BIN --version 2>&1)"
+PY_MINOR=$(sed -e 's/\(Python \)\([[:digit:]]\)\.\([[:digit:]]\)\(.*\)/\3/g' <<< "$PY_VERSION")
+PY_LIB_PATH="/usr/local/lib/python$PY_MAJOR.$PY_MINOR/site-packages"
+echo "PY_LIB_PATH: $PY_LIB_PATH"
+
+export CCACHE_DIR='/ccache' CC="ccache ${CC:-gcc}" CXX="ccache ${CXX:-g++}"
+### Diagnostics ###
+
+# Builder DebianSystemFBCodeBuilder(google/googletest:cmake_defines={u'BUILD_GTEST': u'ON', u'BUILD_SHARED_LIBS': u'OFF'}, gcc_version=u'5', facebook/zstd:git_hash=ShellQuoted(u'$(git describe --abbrev=0 --tags origin/master)'), hyperic/sigar:autoconf_options={u'CFLAGS': u'-fgnu89-inline'}, openr/build:cmake_defines={u'ADD_ROOT_TESTS': u'OFF'}, prefix=u'/usr/local', projects_dir=u'/usr/local/src', google/googletest:git_hash=u'release-1.8.1', wangle/wangle/build:cmake_defines={u'BUILD_TESTS': u'OFF'}, thom311/libnl:git_hash=u'libnl3_2_25', jedisct1/libsodium:git_hash=u'stable', ccache_dir=u'/ccache', make_parallelism=4, no1msd/mstch:git_hash=ShellQuoted(u'$(git describe --abbrev=0 --tags)'), zeromq/libzmq:git_hash=u'v4.2.5')
+hostname
+cat /etc/issue || echo no /etc/issue
+g++ --version || echo g++ not installed
+cmake --version || echo cmake not installed
+
+### Check out facebook/folly, workdir _build ###
+
+mkdir -p '/usr/local/src' && cd '/usr/local/src'
+if [[ ! -d '/usr/local/src'/'folly' ]]; then 
+	git clone https://github.com/'facebook/folly'
+fi
+mkdir -p '/usr/local/src'/'folly'/'_build' && cd '/usr/local/src'/'folly'/'_build'
+git checkout '3ceffd7d145be3c85a2aae39f99eb86ea730bdcc'
+
+### Build and install facebook/folly ###
+
+CXXFLAGS="$CXXFLAGS -fPIC" CFLAGS="$CFLAGS -fPIC" cmake -D'BUILD_SHARED_LIBS'='ON' '..'
+make -j '4' VERBOSE=1 
```

Comment:
Don't make asumptions on the cpu#, e.g., at the very least do
```suggestion
make -j "$(nproc)" VERBOSE=1 
```

A better solution could be to define at the top:
```bash
function _make()
{
    make -j "$(nproc)" "$@"
}
```

and then use it in the later invocations as

```suggestion
_make VERBOSE=1 
```

Reply:
You are right, the cores are evaluated on script generation, I guess. What do you think about:
```
#!/bin/bash

MAKEFLAGS+="-j $(shell nproc)"
```

Reply:
That's a good workaround as well yes.

Reply:
See: b7acf86

Reply:
Log from my 2 core VM:
```
...
    terranet-base: ++ nproc
    terranet-base: + export 'MAKEFLAGS= -j 2'
    terranet-base: + MAKEFLAGS=' -j 2'
...
```